ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/docker_entrypoint.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+TARGET_SYNTHEA_DIR=$(readlink -f "$(dirname "${BASH_SOURCE[0]}")")
+readonly TARGET_SYNTHEA_DIR
+
+BFD_END_STATE_PROPERTIES="$TARGET_SYNTHEA_DIR/end_state.properties"
+readonly BFD_END_STATE_PROPERTIES
+
+NUM_REGEX="^[0-9]+$"
+readonly NUM_REGEX
+
+num_generated_benes=100
+generate_future="false"
+num_future_months=0
+
+help() {
+  echo
+  echo "Docker container for running Synthea generation:"
+  echo
+  echo "-n : Specifies the number of beneficiaries to generate. Defaults to 100"
+  echo "-f : Specifies the number of months in the future to generate claim lines. Defaults to 0"
+  echo "-v : Specifies whether the Synthea executable should log to STDOUT or if its logs should be discared. Defaults to unset (non-verbose)"
+  echo "-h : Shows this help text"
+  exit 0
+}
+
+while getopts ":n:f:vh" opt; do
+  case $opt in
+    n)
+      if ! [[ "$OPTARG" =~ $NUM_REGEX ]]; then
+        echo "ERROR, non-numeric specified ($OPTARG)...exiting" >&2
+        exit 1
+      fi
+      num_generated_benes="$OPTARG"
+      ;;
+    f)
+      if ! [[ "$OPTARG" =~ $NUM_REGEX ]]; then
+        echo "ERROR, non-numeric future month value specified ($OPTARG)...exiting" >&2
+        exit 1
+      fi
+      num_future_months="$OPTARG"
+      if [[ "$OPTARG" -gt 0 ]]; then
+        generate_future="true"
+      fi
+      ;;
+    v)
+      export VERBOSE_SYNTHEA_LOGS="true"
+      ;;
+    h)
+      help
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      ;;
+  esac
+  shift
+done
+
+echo "Checking if the output directory is empty before generation..."
+(
+  cd "$TARGET_SYNTHEA_DIR/output"
+  shopt -s nullglob
+  files=(* .*)
+  if ((${#files[@]} != 2)); then
+    # contents of files array is (. ..)
+    echo "The output directory is not empty; ensure the directory is empty before running"
+    exit 1
+  fi
+)
```

Comment:
This should be unnecessary; the synthea script will check this directory for files, and if any are found, will rename it to output-(timestamp) so that the output path is free (and the previous run is archived safely), and then create a new empty output folder. Checkout out prepare-and-run-synthea.py around line 248.

Reply:
Yeah, I am aware of that behavior. The issue is that I intended `output/` to be bind-mounted to the host directly when running this as a Docker container, and if you do that the container cannot rename any bind-mounted directories. If I were to leave the `output/` directory non-empty, the Python script would crash when trying to rename `output/` since it was a mount, not a directory.

I do agree with this being _unnecessary_ though -- this is surprising behavior and the lines here are redundant with respect to the Python script. I think what I will do is update the usage of the container to make the bind-mount at (in the container) `/usr/local/synthea/out` and `mv` the generated output into that bind-mount when generation is complete.

Reply:
Ah I see Docker issue. Yeah your suggestion sounds good, since that issue isnt immediately obvious when just looking at the code here.

Reply:
Yep, it was a Docker issue; at the time I didn't think it would make sense to have another "output" directory but I think now it makes things much simpler. 

I've gone ahead and made the change I mentioned above (switching the bind-mount from `output/` to `out/`).